ui/app/controllers/jobs/index.js

```diff
@@ -28,11 +51,95 @@ export default Controller.extend(Sortable, Searchable, {
   fuzzySearchProps: computed(() => ['name']),
   fuzzySearchEnabled: true,
 
+  facetOptionsType: computed(() => [
+    { key: 'batch', label: 'Batch' },
+    { key: 'parameterized', label: 'Parameterized' },
+    { key: 'periodic', label: 'Periodic' },
+    { key: 'service', label: 'Service' },
+    { key: 'system', label: 'System' },
+  ]),
+
+  facetOptionsStatus: computed(() => [
+    { key: 'pending', label: 'Pending' },
+    { key: 'running', label: 'Running' },
+    { key: 'dead', label: 'Dead' },
+  ]),
+
+  facetOptionsDatacenter: computed('visibleJobs.[]', function() {
+    const flatten = (acc, val) => acc.concat(val);
+    const allDatacenters = new Set(
+      this.get('visibleJobs')
+        .mapBy('datacenters')
+        .reduce(flatten, [])
+    );
+
+    // Remove any invalid datacenters from the query param/selection
+    const availableDatacenters = Array.from(allDatacenters).compact();
+    scheduleOnce('actions', () => {
+      this.set(
+        'qpDatacenter',
+        qpSerialize(intersection(availableDatacenters, this.get('facetSelectionDatacenter')))
+      );
+    });
+
+    return availableDatacenters.sort().map(dc => ({ key: dc, label: dc }));
+  }),
+
+  facetOptionsPrefix: computed('visibleJobs.[]', function() {
+    // A prefix is defined as the start of a job name up to the first - or .
+    // ex: mktg-analytics -> mktg, ds.supermodel.classifier -> ds
+    const hasPrefix = /.[-._]/;
+
+    // Collect and count all the prefixes
+    const allNames = this.get('visibleJobs').mapBy('name');
+    const nameHistogram = allNames.reduce((hist, name) => {
+      if (hasPrefix.test(name)) {
+        const prefix = name.match(/(.+?)[-.]/)[1];
+        hist[prefix] = hist[prefix] ? hist[prefix] + 1 : 1;
+      }
+      return hist;
+    }, {});
```

Comment:
I'm rubbish at RegEx so not sure on this one, but is there a way to test and match the prefix in one shot?